### PR TITLE
chore: Merge scripts in test/e2e2 and hack/demo-env

### LINKED ISSUE #244: Merge scripts in test/e2e2 and hack/demo-env
### Is your task related to a problem? Please describe.

Many of the scripts in `test/e2e2` and `hack/demo-env` feel (and probably are) redundant, as they do the same things:

* Setup a vcluster environment and install Argo CD in various degrees of completeness
* Start the principal from the current code base
* Start the agent(s) from the current code base

That makes changes to one or the other cumbersome, and both scripts will slowly drift apart.

### Describe the solution you'd like

It would be great if we can use one single set of scripts for both scenarios, e2e tests and the demo env.

### Describe alternatives you've considered

n/a

### Additional context

n/a

### PR DESCRIPTION
**Pull request by jopit**
**What does this PR do / why we need it**:

Merges the contents of the `test-env` and `demo-env` directories into a new directory called `dev-env`

**Which issue(s) this PR fixes**:

Fixes #244

### CODE REVIEW

**Review by jannfis — COMMENTED** [review]
I tested this locally to setup an environment, and it failed on me:

```
-> Generate server.secretkey for agent's argocd-secrets
-> Generate admin password for the control plane
Warning: Value 16 is outside of the allowed entropy range, adjusting it.
-> Create Argo CD on control plane
  --> Creating instance in vcluster control-plane
namespace/argocd created
error: trouble configuring builtin PatchTransformer with config: `
path: argocd-secret.yaml
`: unable to parse SM or JSON patch from [path: "argocd-secret.yaml"]
error: trouble configuring builtin PatchTransformer with config: `
path: argocd-secret.yaml
`: unable to parse SM or JSON patch from [path: "argocd-secret.yaml"]
Switched to context "default".
```

File: hack/dev-env/setup-vcluster-env.sh [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+    # Generate the server secret key for the argocd running on the managed and autonomous agent clusters
+    echo "-> Generate server.secretkey for agent's argocd-secrets"
+    if ! pwmake=$(which pwmake); then
+        pwmake=$(which pwgen)
+    fi
+    echo "data:" >> $TMP_DIR/agent-managed/argocd-secret.yaml
+    echo "  server.secretkey: $($pwmake 56 | base64)" >> $TMP_DIR/agent-managed/argocd-secret.yaml
+    echo "data:" >> $TMP_DIR/agent-autonomous/argocd-secret.yaml
+    echo "  server.secretkey: $($pwmake 56 | base64)" >> $TMP_DIR/agent-autonomous/argocd-secret.yaml
```
Comment (jannfis) [inline hack/dev-env/setup-vcluster-env.sh]:
```suggestion
    echo "data:" >> $TMP_DIR/agent-managed/argocd-secret.yaml
    echo "  server.secretkey: $($pwmake 56 | base64 -w0)" >> $TMP_DIR/agent-managed/argocd-secret.yaml
    echo "data:" >> $TMP_DIR/agent-autonomous/argocd-secret.yaml
    echo "  server.secretkey: $($pwmake 56 | base64 -w0)" >> $TMP_DIR/agent-autonomous/argocd-secret.yaml
```

File: hack/dev-env/setup-vcluster-env.sh [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+    # Generate the argocd admin password for the control plane
+    if [[ "$gen_admin_pwd" == "true" ]]; then
+        echo "-> Generate admin password for the control plane"
+        ADMIN_PASSWORD=$($pwmake 16)
+        password=$(htpasswd -nb -B a $ADMIN_PASSWORD | cut -c 3-)
+        echo "data:" >> $TMP_DIR/control-plane/argocd-secret.yaml
+        echo "  admin.password: $(echo $password | base64)" >> $TMP_DIR/control-plane/argocd-secret.yaml
+        echo "  admin.passwordMtime: $(date +%FT%T%Z | base64)" >> $TMP_DIR/control-plane/argocd-secret.yaml
```
Comment (jannfis) [inline hack/dev-env/setup-vcluster-env.sh]:
```suggestion
        echo "  admin.password: $(echo $password | base64 -w0)" >> $TMP_DIR/control-plane/argocd-secret.yaml
        echo "  admin.passwordMtime: $(date +%FT%T%Z | base64 -w0)" >> $TMP_DIR/control-plane/argocd-secret.yaml
```

File: hack/dev-env/setup-vcluster-env.sh (hunk 1 above)
Comment (jannfis) [reply hack/dev-env/setup-vcluster-env.sh]: -w0 is to not add line wraps to base64 encoded output.

**Review by jannfis — COMMENTED** [review]
Solution as proposed change.

File: hack/dev-env/setup-vcluster-env.sh (hunk 1 above)
Comment (jopit) [reply hack/dev-env/setup-vcluster-env.sh]: Seems that macOS base64 doesn't break lines by default, and doesn't support the `-w0` option.  Have to add some logic to detect if we're on mac and tweak the command accordingly.

Comment (jannfis) [reply hack/dev-env/setup-vcluster-env.sh]: The wonderful world of BSD vs. GNU :)

Comment (jannfis) [reply hack/dev-env/setup-vcluster-env.sh]: Maybe we could just use `base64 | tr -d '\n'`?

Comment (jopit) [reply hack/dev-env/setup-vcluster-env.sh]: I think detecting the OS is cleaner and clearer

Comment (jopit) [reply hack/dev-env/setup-vcluster-env.sh]: Fixed

File: hack/dev-env/setup-vcluster-env.sh (hunk 2 above)
Comment (jopit) [reply hack/dev-env/setup-vcluster-env.sh]: Fixed

**Review by jannfis — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**